python/ray/dag/compiled_dag_node.py

```diff
@@ -530,9 +530,29 @@ def __init__(self):
                 super().__init__(daemon=True)
                 self.in_teardown = False
 
-            def teardown(self):
+            def wait_teardown(self):
+                for ref in outer.worker_task_refs:
+                    try:
+                        ray.get(ref, timeout=10)
+                    except ray.exceptions.GetTimeoutError:
+                        logger.warn(
+                            "At least one actor in the DAG is still running 10s "
+                            "after teardown(). Teardown may hang."
+                        )
+                    except Exception:
```

Comment:
what is this exception catch for? can you comment here & after when this happens?